plugin/storage/es/factory.go

```diff
@@ -221,21 +221,18 @@ func createSpanReader(
 		return nil, fmt.Errorf("--es.use-ilm must always be used in conjunction with --es.use-aliases to ensure ES writers and readers refer to the single index mapping")
 	}
 	return esSpanStore.NewSpanReader(esSpanStore.SpanReaderParams{
-		Client:                        clientFn,
-		MaxDocCount:                   cfg.MaxDocCount,
-		MaxSpanAge:                    cfg.MaxSpanAge,
-		IndexPrefix:                   cfg.IndexPrefix,
-		SpanIndexDateLayout:           cfg.IndexDateLayoutSpans,
-		ServiceIndexDateLayout:        cfg.IndexDateLayoutServices,
-		SpanIndexRolloverFrequency:    cfg.GetIndexRolloverFrequencySpansDuration(),
-		ServiceIndexRolloverFrequency: cfg.GetIndexRolloverFrequencyServicesDuration(),
-		TagDotReplacement:             cfg.Tags.DotReplacement,
-		UseReadWriteAliases:           cfg.UseReadWriteAliases,
-		Archive:                       archive,
-		RemoteReadClusters:            cfg.RemoteReadClusters,
-		Logger:                        logger,
-		MetricsFactory:                mFactory,
-		Tracer:                        tp.Tracer("esSpanStore.SpanReader"),
+		Client:              clientFn,
+		MaxDocCount:         cfg.MaxDocCount,
+		MaxSpanAge:          cfg.MaxSpanAge,
+		IndexPrefix:         cfg.IndexPrefix,
+		Indices:             cfg.Indices,
```

Comment:
span reader does not need sampling and dependencies settings. We should not pass more configs than necessary.